pkg/odo/cli/component/devfile.go

```diff
@@ -137,3 +137,40 @@ func getNamespace() (string, error) {
 	Namespace := envInfo.GetNamespace()
 	return Namespace, nil
 }
+
+// DevfileComponentDelete deletes the devfile component
+func (do *DeleteOptions) DevfileComponentDelete() error {
```

Comment:
Something to keep in mind: Rather than having a single devfile.go that contains the functions for push, delete, etc. We should probably separate them out into their own files.

Reply:
I will have a look into it.